runtime-sdk-macros/src/event_derive.rs

```diff
@@ -0,0 +1,154 @@
+use darling::{util::Flag, FromDeriveInput, FromVariant};
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote};
+use syn::{DeriveInput, Ident};
+
+#[derive(FromDeriveInput)]
+#[darling(supports(enum_any), attributes(event))]
+struct Event {
+    ident: Ident,
+
+    data: darling::ast::Data<EventVariant, darling::util::Ignored>,
+
+    /// The type ident of the event enum.
+    module: syn::Path,
```

Comment:
it'd be nice to remove this using type information. not a huge fan of getting it, though